openssl/src/hash.rs

```diff
@@ -16,39 +49,66 @@ use error::ErrorStack;
 pub struct MessageDigest(*const ffi::EVP_MD);
 
 impl MessageDigest {
+    /// Construct a `MessageDigest` that will hash bytes supplied to `Hasher.update` using the MD5
```

Comment:
Let's shorten the first sentences here up a bit - maybe something like "Returns a `MessageDigest` corresponding to the MD5 algorithm".